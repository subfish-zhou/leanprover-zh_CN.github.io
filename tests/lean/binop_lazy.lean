def f : IO Unit :=
  (do IO.println "case 1"; throw (IO.userError "failed"))
  <|>
  (do IO.println "case 2"; throw (IO.userError "failed"))
  <|>
  (do IO.println "case 3")
  <|>
  (let x := dbg_trace "hello"; 1
   IO.println x)

#eval f -- should not print hello
