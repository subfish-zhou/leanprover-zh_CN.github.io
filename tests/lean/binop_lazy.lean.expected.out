case 1
case 2
case 3
